pkg/scheduler/scheduler.go

```diff
@@ -299,8 +324,23 @@ func (s *Scheduler) schedule(ctx context.Context) wait.SpeedSignal {
 			log.V(5).Info("Finished waiting for all admitted workloads to be in the PodsReady condition")
 		}
 		e.status = nominated
+
+		isAdmitted := true
 		if err := s.admit(ctx, e, cq); err != nil {
 			e.inadmissibleMsg = fmt.Sprintf("Failed to admit workload: %v", err)
+			isAdmitted = false
+		}
+
+		if isAdmitted && features.Enabled(features.AdmissionFairSharing) {
+			if err := s.addFairSharingEntryPenalty(ctx, e); err != nil {
+				log.Error(err, "failed to add fair sharing entry penalty")
+			}
```

Comment:
My main concern is that if an error occurs, we may not be able to handle the case again and could lose status changes, since the scheduler runs only once and we are only logging the error.

We can fix Conflict error with patch SSA and strict false like we did on workload but what about other errors?